### PR TITLE
Skip raising exception if grep doesn't found any matching pattern on pod

### PR DESCRIPTION
**Pull request by vavuthu**
Fixes: #6155

Signed-off-by: vavuthu <vavuthu@redhat.com>

### CODE REVIEW

**Comment by vavuthu** [comment]
verified locally
```
20:27:06 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: oc -n openshift-storage rsh session-fedora-pod-9a72a6821a2c47c2a700f grep -i kgf /tmp/list_object
20:27:08 - MainThread - ocs_ci.utility.utils - WARNING  - Command stderr: command terminated with exit code 1

20:27:08 - MainThread - ocs_ci.utility.utils - INFO  - No results found for grep
20:27:08 - MainThread - tests.test1 - INFO  - 
PASSED

========================================================================================== 1 passed in 73.70s (0:01:13) ==========================================================================================
```

File: ocs_ci/utility/utils.py [hunk 1]
```diff
@@ -609,10 +609,16 @@ def exec_cmd(
...
     if completed_process.returncode and not ignore_error:
-        raise CommandFailed(
-            f"Error during execution of command: {masked_cmd}."
-            f"\nError is {masked_stderr}"
-        )
+        if (
+            "grep" in masked_cmd
+            and b"command terminated with exit code 1" in completed_process.stderr
+        ):
+            log.info("No results found for grep")
```
Comment (ebenahar) [inline ocs_ci/utility/utils.py:616]:
```suggestion
            log.info(f"No results found for {masked_cmd}")
```

Comment (vavuthu) [reply ocs_ci/utility/utils.py]: Done

File: ocs_ci/utility/utils.py [hunk 2]
```diff
@@ -609,10 +609,16 @@ def exec_cmd(
...
     if completed_process.returncode and not ignore_error:
-        raise CommandFailed(
-            f"Error during execution of command: {masked_cmd}."
-            f"\nError is {masked_stderr}"
-        )
+        if (
+            "grep" in masked_cmd
+            and b"command terminated with exit code 1" in completed_process.stderr
+        ):
+            log.info(f"No results found for {masked_cmd}")
```
Comment (petr-balogh) [inline ocs_ci/utility/utils.py:616]: Add something to highlight Grep here.

And document this in the docstring.

**Review by petr-balogh — APPROVED** [review]

**Review by Akarsha-rai — APPROVED** [review]

**Comment by vavuthu** [comment]
re-verified locally:

```
15:45:35 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: oc -n openshift-storage rsh session-fedora-pod-9a72a6821a2c47c2a700f grep -i 'An error occurred (504) when calling the ListObjectsV2 operation (reached max retries: 2): Gateway Timeout' /tmp/list_object
15:45:38 - MainThread - ocs_ci.utility.utils - WARNING  - Command stderr: command terminated with exit code 1

15:45:38 - MainThread - ocs_ci.utility.utils - INFO  - No results found for grep command: oc -n openshift-storage rsh session-fedora-pod-9a72a6821a2c47c2a700f grep -i 'An error occurred (504) when calling the ListObjectsV2 operation (reached max retries: 2): Gateway Timeout' /tmp/list_object 
PASSED
```

**Pull request merged**